shared/src/test/scala/com/paulbutcher/test/mock/MockTest.scala

```diff
@@ -488,5 +488,40 @@ class MockTest extends AnyFreeSpec with MockFactory with Matchers {
 
       "stub[HasNotifyMethod]" should compile
     }
+
+    "mock generic constructor arguments" in {
+      class WithOption(opt: Option[String])
+      class WithInt(i: Int)
+      class WithString(s: String)
+      "mock[WithOption]" should compile
+      "mock[WithInt]" should compile
+      "mock[WithString]" should compile
+
+    }
+
+    "mock type constructor arguments" in {
+      class WithTC[TC[_]](tc: TC[Int])
+      type ID[A] = A
+      "mock[WithTC[List]]" should compile
+      pendingUntilFixed {
+        "mock[WithTC[ID]]" should compile
```

Comment:
this still fails, it doesn't notice that this ends up being an Int needing a cast.

Reply:
I think it is fine